cpp/tests/utilities/compare_column_wrappers.cuh

```diff
@@ -59,6 +59,24 @@ enum : bool {
     dont_print_unequal_pairs = false,
 };
 
+template<class T>
+typename std::enable_if_t<std::is_floating_point<T>::value, 
+bool> almost_equal(T x, T y, int ulp)
+{
+  // the machine epsilon has to be scaled to the magnitude of the values used
+  // and multiplied by the desired precision in ULPs (units in the last place)
+  return std::abs(x-y) <= std::numeric_limits<T>::epsilon() * std::abs(x+y) * ulp
+    // unless the result is subnormal
+    || std::abs(x-y) < std::numeric_limits<T>::min();
+}
+
+template<class T>
+typename std::enable_if_t<!std::is_floating_point<T>::value, 
+bool> almost_equal(T x, T y, int)
+{
+  return x == y;
```

Comment:
This doesn't look like "almost" equal...

Reply:
This is an overload for non-floating point types, which are all integer based anyway.